src/core/internal/string.d

```diff
@@ -28,27 +28,24 @@ Params:
 Returns:
     The unsigned integer value as a string of characters
 */
-char[] unsignedToTempString()(ulong value, return scope char[] buf, uint radix = 10) @safe
+char[] unsignedToTempString(uint radix = 10)(ulong value, return scope char[] buf) @safe
+if (radix >= 2 && radix <= 16)
 {
-    if (radix < 2)
-        // not a valid radix, just return an empty string
-        return buf[$ .. $];
-
     size_t i = buf.length;
     do
     {
+        uint x = void;
         if (value < radix)
         {
-            ubyte x = cast(ubyte)value;
-            buf[--i] = cast(char)((x < 10) ? x + '0' : x - 10 + 'a');
-            break;
+            x = cast(uint)value;
+            value = 0;
```

Comment:
Nitpicking, but is this better than the original that repeated the code, but didn't test value again?

In fact, the while loop could be `while(true)` if there is a break in here, as a value of 0 will never result from the other branch.

Reply:
Good point. Judging by the size of code generated, the finite loop is marginally better: https://godbolt.org/z/G6z3Kj. Add to that the mild inconvenience in source code duplication and it's okay to stick with the current version.